src/hierarchies/lddp_uni_hierarchy.hpp

```diff
@@ -0,0 +1,106 @@
+#ifndef BAYESMIX_HIERARCHIES_LDDP_UNI_HIERARCHY_HPP_
+#define BAYESMIX_HIERARCHIES_LDDP_UNI_HIERARCHY_HPP_
+
+#include <google/protobuf/stubs/casts.h>
+
+#include <Eigen/Dense>
+#include <memory>
+
+#include "../../proto/cpp/hierarchy_prior.pb.h"
+#include "../../proto/cpp/marginal_state.pb.h"
+#include "base_hierarchy.hpp"
+#include "dependent_hierarchy.hpp"
+
+class LDDPUniHierarchy : public DependentHierarchy {
```

Comment:
I know this is used only for the Linear Dependent Dirichlet Process, but in the future this same class might be used in combination with any kind of mixing. Wouldn't `LinRegHierarchy` be a better naming?

Reply:
Absolutely! `LinRegUniHierarchy` it is.